### ISSUE TITLE
can't use shield

### ISSUE THREAD

**Issue by LextYi** [hashalite/Freecam#36]
game crashes when using shield

**Comment by LextYi** [comment]
```
---- Minecraft Crash Report ----
// Don't do that.

Time: 2022/6/20 下午3:53
Description: Ticking entity

java.lang.IllegalArgumentException: Invalid hand null
	at Not Enough Crashes deobfuscated stack trace.(1.17.1+build.65)
	at net.minecraft.entity.LivingEntity.getStackInHand(LivingEntity:1897)
	at net.minecraft.entity.LivingEntity.tickActiveItemStack(LivingEntity:2941)
	at net.minecraft.entity.LivingEntity.tick(LivingEntity:2309)
	at net.minecraft.entity.player.PlayerEntity.tick(PlayerEntity:274)
	at net.minecraft.client.network.ClientPlayerEntity.tick(ClientPlayerEntity:210)
	at net.minecraft.client.world.ClientWorld.tickEntity(ClientWorld:179)
	at net.minecraft.world.World.tickEntity(World:483)
	at net.minecraft.client.world.ClientWorld.method_32124(ClientWorld:167)
	at net.minecraft.world.EntityList.forEach(EntityList:54)
	at net.minecraft.client.world.ClientWorld.tickEntities(ClientWorld:163)
	at net.minecraft.client.MinecraftClient.tick(MinecraftClient:1750)
	at net.minecraft.client.MinecraftClient.render(MinecraftClient:1081)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:728)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.client.MinecraftClient.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(MinecraftClient:11608)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:750)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.client.MinecraftClient.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(MinecraftClient:11608)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:750)
	at net.minecraft.client.main.Main.main(Main:217)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_1309.method_5998(class_1309.java:1897)
	at net.minecraft.class_1309.method_6076(class_1309.java:2941)
	at net.minecraft.class_1309.method_5773(class_1309.java:2309)
	at net.minecraft.class_1657.method_5773(class_1657.java:274)
	at net.minecraft.class_746.method_5773(class_746.java:210)
	at net.minecraft.class_638.method_18646(class_638.java:179)

-- Entity being ticked --
Details:
	Entity Type: minecraft:player (net.xolt.freecam.util.FreeCamera)
	Entity ID: 32169
	Entity Name: FreeCamera
	Entity's Exact location: -404.13, 5.44, 137.15
	Entity's Block location: World: (-405,5,137), Section: (at 11,5,9 in -26,0,8; chunk contains blocks -416,0,128 to -401,255,143), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Entity's Momentum: 0.03, 0.02, 0.15
	Entity's Passengers: []
	Entity's Vehicle: null
	Entity NBT: {AbsorptionAmount:0.0f,Air:300s,Attributes:[{Base:0.10000000149011612d,Name:"minecraft:generic.movement_speed"}],Brain:{memories:{}},DataVersion:2730,DeathTime:0s,EnderItems:[],FallDistance:0.0f,FallFlying:0b,Fire:0s,Health:20.0f,HurtByTimestamp:0,HurtTime:0s,Inventory:[],Invulnerable:0b,Motion:[0.02571979799617736d,0.022394880889892575d,0.1501922543138888d],OnGround:0b,PortalCooldown:0,Pos:[-404.1321932167279d,5.4440128573799145d,137.14672403283333d],Rotation:[170.28476f,22.70479f],Score:0,SelectedItemSlot:0,SleepTimer:0s,UUID:[I;-1063202408,810502349,-1221563830,353430609],XpLevel:0,XpP:0.0f,XpSeed:0,XpTotal:0,abilities:{flySpeed:0.1f,flying:1b,instabuild:0b,invulnerable:0b,mayBuild:1b,mayfly:0b,walkSpeed:0.1f},foodExhaustionLevel:0.0f,foodLevel:20,foodSaturationLevel:5.0f,foodTickTimer:0}
Stacktrace:
	at Not Enough Crashes deobfuscated stack trace.(1.17.1+build.65)
	at java.lang.Thread.getStackTrace(Thread.java:1596)
	at net.minecraft.util.crash.CrashReportSection.trimStackTraceEnd(CrashReportSection:521)
	at net.minecraft.util.crash.CrashReport.addElement(CrashReport:198)
	at net.minecraft.world.World.addDetailsToCrashReport(World:845)
	at net.minecraft.client.world.ClientWorld.addDetailsToCrashReport(ClientWorld:370)
	at net.minecraft.client.MinecraftClient.addDetailsToCrashReport(MinecraftClient:2399)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:747)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.client.MinecraftClient.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(MinecraftClient:11608)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:750)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.client.MinecraftClient.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(MinecraftClient:11608)
	at net.minecraft.client.MinecraftClient.run(MinecraftClient:750)
	at net.minecraft.client.main.Main.main(Main:217)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 2 total; [class_746['LextYi'/31922, l='ClientLevel', x=-404.40, y=4.00, z=135.60], FreeCamera['FreeCamera'/32169, l='ClientLevel', x=-404.13, y=5.44, z=137.15]]
	Chunk stats: 1024, 242
	Level dimension: minecraft:overworld
	Level spawn location: World: (-144,4,64), Section: (at 0,4,0 in -9,0,4; chunk contains blocks -144,0,64 to -129,255,79), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 379925 game time, 6000 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:370)
	at net.minecraft.class_310.method_1587(class_310.java:2399)
	at net.minecraft.class_310.method_1514(class_310.java:747)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.class_310.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(class_310.java:11608)
	at net.minecraft.class_310.method_1514(class_310.java:750)
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37)
	at net.minecraft.class_310.modify$bhi000$atTheEndOfFirstCatchBeforePrintingCrashReport(class_310.java:11608)
	at net.minecraft.class_310.method_1514(class_310.java:750)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: Default, Fabric Mods

-- System Details --
Details:
	Minecraft Version: 1.17.1
	Minecraft Version ID: 1.17.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 666728056 bytes (635 MiB) / 1342177280 bytes (1280 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: 11th Gen Intel(R) Core(TM) i5-11400F @ 2.60GHz
	Identifier: Intel64 Family 6 Model 167 Stepping 1
	Microarchitecture: unknown
	Frequency (GHz): 2.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GTX 1650
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x1f82
	Graphics card #0 versionInfo: DriverVersion=30.0.14.9729
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 3.20
	Memory slot #0 type: DDR4
	Virtual memory max (MB): 19837.44
	Virtual memory used (MB): 16534.15
	Swap memory total (MB): 11776.00
	Swap memory used (MB): 1323.52
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		advanced_runtime_resource_pack: Runtime Resource Pack 0.4.3
		baritone: Baritone 1.7.2
		betterbeds: Better Beds 1.2.0
		cleancut: CleanCut 1.17-4.0-fabric
		clientcommands: Client Commands 2.5.5
		clientcommands-scripting: Clientcommands Scripting 1.2
		cloth-basic-math: cloth-basic-math 0.6.1
		cloth-config2: Cloth Config v5 5.3.63
		com_github_hube12_seed: SEED master-SNAPSHOT
		com_velocitypowered_velocity-native: velocity-native 1.1.0-SNAPSHOT
		crowdin-translate: CrowdinTranslate 1.3+1.17
		cullleaves: Cull Leaves 2.3.2
		enhancedblockentities: Enhanced Block Entities 0.5+1.17
		entityculling: EntityCulling-Fabric 1.5.0
		fabric: Fabric API 0.46.1+1.17
		fabric-api-base: Fabric API Base 0.4.0+cf39a74318
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.5.0+e821752d18
		fabric-biome-api-v1: Fabric Biome API (v1) 3.2.2+cf39a74318
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.6+cf39a74318
		fabric-command-api-v1: Fabric Command API (v1) 1.1.4+cf39a74318
		fabric-commands-v0: Fabric Commands (v0) 0.2.3+cf39a74318
		fabric-containers-v0: Fabric Containers (v0) 0.1.13+cf39a74318
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.1+3447790d18
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.6+cf39a74318
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.15+b556f28c18
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.1+377137cc18
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.12+e99fbe1218
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.4+cf39a74318
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.8+cf39a74318
		fabric-item-api-v1: Fabric Item API (v1) 1.3.0+6617390918
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.2+cf39a74318
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.6+2a2bb57318
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.4+cf39a74318
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.6+0392f3a618
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.5+cf39a74318
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.3+cf39a74318
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.7+cf39a74318
		fabric-models-v0: Fabric Models (v0) 0.3.1+cf39a74318
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.14+cf39a74318
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.12+cf39a74318
		fabric-networking-v0: Fabric Networking (v0) 0.3.3+cf39a74318
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.1+f907116918
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.8+cf39a74318
		fabric-particles-v1: Fabric Particles (v1) 0.2.5+cf39a74318
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.8.0+ea29b33318
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.6+ce16ae5918
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.10+ce16ae5918
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.6+cf39a74318
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.6+cf39a74318
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.2.1+cf39a74318
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.7+cf39a74318
		fabric-rendering-v1: Fabric Rendering (v1) 1.10.1+377137cc18
		fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 1.0.1+591a0f2118
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.10+f09604ce18
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.5+cf39a74318
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.9+cf39a74318
		fabric-structure-api-v1: Fabric Structure API (v1) 1.2.0+48a9ad8918
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.3+cf39a74318
		fabric-textures-v0: Fabric Textures (v0) 1.0.7+cf39a74318
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.3+3b96517518
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.4+cf39a74318
		fabricloader: Fabric Loader 0.14.8
		fastopenlinksandfolders: FastOpenResourcePacks 1.1.1
		forgetmechunk: Forget Me Chunk 1.17.X-1.0.3
		freecam: Freecam 1.0.8
		hydrogen: Hydrogen 0.3
		imblocker: IMBlockerFabric 1.0.17
		indium: Indium 1.0.1+mc1.17.1
		inventory_control_tweaks: Inventory Control Tweaks 1.3.9+MC1.17-1.17.1
		inventoryprofilesnext: Inventory Profiles Next 1.5.3
		java: OpenJDK 64-Bit Server VM 16
		krypton: Krypton 0.1.4
		languagereload: Language Reload 1.2
		lazy-language-loader: lazy-language-loader 0.2.2
		lazydfu: LazyDFU 0.1.2
		lithium: Lithium 0.7.5
		memoryleakfix: memory Leak Fix 1.17.1-0.1.1
		midnightlib: MidnightLib 0.2.9
		minecraft: Minecraft 1.17.1
		modmenu: Mod Menu 2.0.14
		mousetweaks: Mouse Tweaks 2.15
		multiconnect-api: MultiConnect API 1.4
		notenoughcrashes: Not Enough Crashes 4.1.4+1.17.1
		org_joml_joml: joml 1.10.2
		recipecache: recipecache 0.2.1-1.17.1
		seedcrackerx: SeedCrackerX 2.10.1
		seedcrackerx-api: SeedCrackerX API 2.10.1
		smoothboot: Smooth Boot 1.16.5-1.6.0
		sodium: Sodium 0.3.4+build.13
		starlight: Starlight 1.0.0+fabric.73f6d37
	Launched Version: fabric-loader-0.14.8-1.17.1
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: NVIDIA GeForce GTX 1650/PCIe/SSE2 GL version 3.2.0 NVIDIA 497.29, NVIDIA Corporation
	Window size: 1920x1027
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: id=1280, source=API, type=ERROR, severity=HIGH, message='GL_INVALID_ENUM error generated. Polygon modes for <face> are disabled in the current profile.' x 4
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Integrated Server (map_client.txt)
	Graphics mode: fast
	Resource Packs: vanilla, Fabric Mods
	Current Language: English (US)
	CPU: 12x 11th Gen Intel(R) Core(TM) i5-11400F @ 2.60GHz
	Player Count: 1 / 8; [class_3222['LextYi'/31922, l='ServerLevel[TEST]', x=-404.40, y=4.00, z=135.60]]
	Data Packs: vanilla, Fabric Mods, file/Donkey Dupe (incompatible)
	Client Crashes Since Restart: 3
	Integrated Server Crashes Since Restart: 0
	Suspected Mods: Minecraft (minecraft), Fabric Loader (fabricloader), Not Enough Crashes (notenoughcrashes)
```

**Comment by hashalite** [comment]
This has been fixed in the next update.
